### PR TITLE
Add Japanese translation for kubectl

### PR DESCRIPTION
**Pull request by girikuncoro**
**What this PR does / why we need it**:
This PR provides first attempt to translate kubectl in Japanese (related to #40645 and #40591). 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
No issues

**Special notes for your reviewer**:
Should be reviewed by member of Japanese k8s community (I stayed in Japan for 4 years, but my language is not as good as native Japanese)

### CODE REVIEW

**Comment by girikuncoro** [comment]
I've just sign up the CLA

**Comment by soltysh** [comment]
@kubernetes/sig-cli-pr-reviews do we have someone who knows Japanese to verify this?

**Comment by brendandburns** [comment]
Calling @IanLewis who knows Japanese afaik...

**Comment by brendandburns** [comment]
@k8s-bot ok to test

**Comment by brendandburns** [comment]
@girikuncoro Thanks for the contribution!

Spot checks via Google Translate look ok.  We'll wait a couple days to see if another Japanese speaker can do a check, then I'll LGTM.

**Comment by vfreex** [comment]
Shouldn't the language code of Japanese be ja_JP?

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/use_context.go#L48
+msgid "Sets the current-context in a kubeconfig file"
+msgstr "kubeconfigに現在のコンテキストを設定する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/unset.go#L47
+msgid "Unsets an individual value in a kubeconfig file"
+msgstr "kubeconfigに単一値の設定を削除する"
+
+msgid "Update the annotations on a resource"
+msgstr "リソースでのあのテーショーンを更新する"
```
Comment (superbrothers) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: リソースのアノテーションを更新する

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+msgid ""
+"watch is only supported on individual resources and resource collections - "
+"%d resources were found"
+msgid_plural ""
+"watch is only supported on individual resources and resource collections - "
+"%d resources were found"
+msgstr[0] ""
+"watchは単一リソース及びリソースコレクションでのみサポートされている - "
+"リソース　%d　が見つかった"
```
Comment (superbrothers) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: %d個のリソースが見つかりました

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+msgstr[1] ""
+"watchは単一リソース及びリソースコレクションでのみサポートされている - "
+"リソース　%d　が見つかった"
```
Comment (superbrothers) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: %d個のリソースが見つかりました

File: pkg/util/i18n/i18n.go [hunk 4]
```diff
@@ -35,6 +35,7 @@ var knownTranslations = map[string][]string{
 		"default",
 		"en_US",
 		"fr_FR",
+		"jp_JP",
```
Comment (TakanariKo) [inline pkg/util/i18n/i18n.go]: jp_JP

Comment (vfreex) [inline pkg/util/i18n/i18n.go]: The [ISO 639 language code of Japanese language](https://www.loc.gov/standards/iso639-2/php/code_list.php) is `ja`, so this should be changed to `ja_JP`.

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+msgid "Sets a user entry in kubeconfig"
+msgstr "kubeconfigにユーザーエントリを設定する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/set.go#L59
+msgid "Sets an individual value in a kubeconfig file"
+msgstr "kubeconfigファイルに単一値を設定する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/use_context.go#L48
+msgid "Sets the current-context in a kubeconfig file"
+msgstr "kubeconfigに現在のコンテキストを設定する"
```
Comment (superbrothers) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: kubeconfigにカレントコンテキストを設定する

**Review by vfreex — CHANGES_REQUESTED** [review]

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+"PO-Revision-Date: 2017-01-29 22:54-0800\n"
+"Last-Translator: Giri Kuncoro <girikuncoro@gmail.com>\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"X-Generator: Poedit 1.6.10\n"
+"X-Poedit-SourceCharset: UTF-8\n"
+"Language-Team: \n"
+"Plural-Forms: nplurals=2; plural=(n > 1);\n"
+"Language: jp\n"
```
Comment (superbrothers) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: ja

**Comment by girikuncoro** [comment]
Thanks for the reviews and comments! I'll wait until tomorrow in case more comments come up, and make changes accordingly tomorrow (Pacific Time)

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+msgid "Display clusters defined in the kubeconfig"
+msgstr "kubeconfigで定義されたクラスターを表示する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/view.go#L64
+msgid "Display merged kubeconfig settings or a specified kubeconfig file"
+msgstr "kubeconfig設定または指示したkubeconfigファイルを表示する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/current_context.go#L48
+msgid "Displays the current-context"
+msgstr "現在のコンテキストを表示する"
```
Comment (superbrothers) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: カレントコンテキストを表示する

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+msgid "Describe one or many contexts"
+msgstr "１つまたは複数のコンテキストを記述する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/get_clusters.go#L40
+msgid "Display clusters defined in the kubeconfig"
+msgstr "kubeconfigで定義されたクラスターを表示する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/view.go#L64
+msgid "Display merged kubeconfig settings or a specified kubeconfig file"
+msgstr "kubeconfig設定または指示したkubeconfigファイルを表示する"
```
Comment (superbrothers) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: マージされたkubeconfigの設定または指定されたkubeconfigファイルを表示する

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+msgstr "リソースでのあのテーショーンを更新する"
+
+msgid ""
+"watch is only supported on individual resources and resource collections - "
+"%d resources were found"
+msgid_plural ""
+"watch is only supported on individual resources and resource collections - "
+"%d resources were found"
+msgstr[0] ""
+"watchは単一リソース及びリソースコレクションでのみサポートされている - "
```
Comment (superbrothers) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: watchは単一リソース及びリソースコレクションのみサポートしています

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+"リソース　%d　が見つかった"
+msgstr[1] ""
+"watchは単一リソース及びリソースコレクションでのみサポートされている - "
```
Comment (superbrothers) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: watchは単一リソース及びリソースコレクションのみサポートしています

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+msgid "Delete the specified cluster from the kubeconfig"
+msgstr "kubeconfigから指定したクラスターを削除する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/delete_context.go#L38
+msgid "Delete the specified context from the kubeconfig"
+msgstr "kubeconfigから指定したコンテキストを削除する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/get_contexts.go#L62
+msgid "Describe one or many contexts"
+msgstr "１つまたは複数のコンテキストを記述する"
```
Comment (superbrothers) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: 1つまたは複数のコンテキストを記述する

**Comment by superbrothers** [comment]
I reviewed the translation. Nice PR! 👍

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po (hunk 11 above)
Comment (superbrothers) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: I prefer to use Hankaku (half-width) for numbers.

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+"Content-Transfer-Encoding: 8bit\n"
+"X-Generator: Poedit 1.6.10\n"
+"X-Poedit-SourceCharset: UTF-8\n"
+"Language-Team: \n"
+"Plural-Forms: nplurals=2; plural=(n > 1);\n"
+"Language: jp\n"
+
+# https://github.com/kubernetes/kubernetes/blob/masterpkg/kubectl/cmd/apply.go#L98
+msgid "Apply a configuration to a resource by filename or stdin"
+msgstr "ファイル名または標準入力でリソースにコンフィグを適用する"
```
Comment (chumaltd) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: ファイル名を指定または標準入力経由でリソースにコンフィグを適用する

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+msgid "Sets a context entry in kubeconfig"
+msgstr "kubeconfigにコンテキストエントリを設定する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/create_authinfo.go#L103
+msgid "Sets a user entry in kubeconfig"
+msgstr "kubeconfigにユーザーエントリを設定する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/set.go#L59
+msgid "Sets an individual value in a kubeconfig file"
+msgstr "kubeconfigファイルに単一値を設定する"
```
Comment (chumaltd) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: kubeconfigファイル内の変数を個別に設定する

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/use_context.go#L48
+msgid "Sets the current-context in a kubeconfig file"
+msgstr "kubeconfigに現在のコンテキストを設定する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/unset.go#L47
+msgid "Unsets an individual value in a kubeconfig file"
+msgstr "kubeconfigに単一値の設定を削除する"
```
Comment (chumaltd) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: kubeconfigファイルから変数を個別に削除する

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po (hunk 13 above)
Comment (chumaltd) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: if "an individual value" is some technical term, there may be better translation. but I don't see "単一値" in everyday life.

File: translations/kubectl/jp_JP/LC_MESSAGES/k8s.po [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+msgid "Display merged kubeconfig settings or a specified kubeconfig file"
+msgstr "kubeconfig設定または指示したkubeconfigファイルを表示する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/current_context.go#L48
+msgid "Displays the current-context"
+msgstr "現在のコンテキストを表示する"
+
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/config/config.go#L39
+msgid "Modify kubeconfig files"
+msgstr "kubeconfigファイルを編集する"
```
Comment (chumaltd) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: kubeconfigファイルを変更する

Comment (chumaltd) [inline translations/kubectl/jp_JP/LC_MESSAGES/k8s.po]: I think "変更" is more suitable for atomic change nuance. "編集" has emphasis on authoring aspect like with vi.

**Comment by girikuncoro** [comment]
@superbrothers @chumaltd @vfreex @TakanariKo I made changes, can you check?

**Comment by girikuncoro** [comment]
@brendandburns for some reason my jenkins bazel build and verification are failing, any guidance on this?

for bazel build I might need to clean up something (this was passing in my first push)
```
W0511 20:33:28.654] fatal: Unable to create '/root/.cache/git/k8s.io/kubernetes/index.lock': File exists.
```

for jenkins verification I have no idea what went wrong in my changes

**Comment by superbrothers** [comment]
lgtm 👍

**Comment by chumaltd** [comment]
lgtm. Thanks for great work!

File: translations/kubectl/ja_JP/LC_MESSAGES/k8s.po [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"X-Generator: Poedit 1.6.10\n"
+"X-Poedit-SourceCharset: UTF-8\n"
+"Language-Team: \n"
+"Plural-Forms: nplurals=2; plural=(n > 1);\n"
+"Language: ja\n"
+
+# https://github.com/kubernetes/kubernetes/blob/masterpkg/kubectl/cmd/apply.go#L98
```
Comment (shiywang) [inline translations/kubectl/ja_JP/LC_MESSAGES/k8s.po]: Seems those url all pointing to the wrong line. Does that matters ? also I saw there's a `template.pot` file. Does this file auto-generate content when command changed ?  @brendandburns

Comment (brendandburns) [inline translations/kubectl/ja_JP/LC_MESSAGES/k8s.po]: I can update the script.

template.pot does not automatically get updated (yet) but this PR shouldn't affect `template.pot`

**Comment by brendandburns** [comment]
@k8s-bot bazel test this
@k8s-bot verify test this

**Comment by brendandburns** [comment]
Ah, @girikuncoro you need to run `./hack/generate-bindata.sh`

Sorry about that...

**Comment by girikuncoro** [comment]
@brendandburns ready for lgtm now, thanks for the suggestion 😄

**Comment by warmchang** [comment]
It's look like ready to lgtm & merge, and then I'll commit the Simplified Chinese translation PR.

**Comment by shiywang** [comment]
@warmchang sorry, I think I already made that https://github.com/kubernetes/kubernetes/pull/45573, you could have a review if you want : )

**Comment by warmchang** [comment]
@shiywang , glad to see your PR and good translation! Maybe i should submit the Traditional Chinese version. :smile:

**Comment by brendandburns** [comment]
/lgtm
/approve

THanks again.

**Comment by brendandburns** [comment]
@girikuncoro Sorry, this now needs a rebase...

Appologies I will shepard this in asap once it's rebased...

**Comment by warmchang** [comment]
@girikuncoro , 2565 files changed, It look like better to rebase the PR.

**Comment by shiywang** [comment]
seems you "rebased" too much

**Comment by girikuncoro** [comment]
@brendandburns sorry my bad, I messed up the PR. I've moved it to the new one at #46756. closing this for now

**Comment by girikuncoro** [comment]
@shiywang my bad, accidentally pushed some mistake 😓  moved to #46756

**Pull request closed without merging**